### ISSUE TITLE
unable to backward propagate convolution with padded input

### ISSUE THREAD

**Issue by jay746** [pytorch/pytorch#60075]
## 🐛 Bug

In a graph where input is padded with required zeros and then passed to do convolution and then backward is called. 
I see test failing with error:
`>>> res.backward(grad_in)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jthakur/vnv_syn_new/lib/python3.6/site-packages/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/jthakur/vnv_syn_new/lib/python3.6/site-packages/torch/autograd/__init__.py", line 147, in backward
    allow_unreachable=True, accumulate_grad=True)  # allow_unreachable flag
RuntimeError: could not create a primitive`

## To Reproduce
if in_channels < 105 then  backpropagation is working fine.

```
import torch

in_channels = 105
out_channels = 1
ifm_shape = [1, in_channels, 1, 1]
padding = [8, 5, 9, 4]
groups = 1
kernel_size = [1, 1]
dilation = [1, 1]
stride = [14, 14]

ifm = torch.rand(ifm_shape)
ifm.requires_grad_(True)
padded_ifm = torch.nn.functional.pad(ifm, padding)

op = torch.nn.Conv2d(
    in_channels=in_channels,
    out_channels=out_channels,
    kernel_size=kernel_size,
    stride=stride,
    dilation=dilation,
    groups=groups,
)

res = op(padded_ifm)

grad_in = torch.rand(res.shape)

res.backward(grad_in)
```

## Environment

 - PyTorch Version (e.g., 1.0): 1.8.1
 - OS (e.g., Linux): Linux 
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): NA
 - Python version: 3.6.9
 - CUDA/cuDNN version: CPU
 - GPU models and configuration: NA
 - Any other relevant information: NA

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @VitalyFedyunin @albanD @mruberry @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7

**Comment by albanD** [comment]
Hi,

Could you share a self-contained code sample that we can use to reproduce this please?

**Comment by jay746** [comment]
@albanD ,
Yes reproduction code is already mentioned in bug info.
Anyway adding again.

`import torch

in_channels = 105
out_channels = 1
ifm_shape = [1, in_channels, 1, 1]
padding = [8, 5, 9, 4]
groups = 1
kernel_size = [1, 1]
dilation = [1, 1]
stride = [14, 14]

ifm = torch.rand(ifm_shape)
ifm.requires_grad_(True)
padded_ifm = torch.nn.functional.pad(ifm, padding)

op = torch.nn.Conv2d(
in_channels=in_channels,
out_channels=out_channels,
kernel_size=kernel_size,
stride=stride,
dilation=dilation,
groups=groups,
)

res = op(padded_ifm)

grad_in = torch.rand(res.shape)

res.backward(grad_in)`

**Comment by albanD** [comment]
Ho sorry I missed that!

The problem seems to come from mkldnn as this error is an internal error there.
You can set `torch.backends.mkldnn.enabled=False` to disable it and have the code run as expected as a temporary workaround.

**Comment by gchanan** [comment]
reproduces on 1.9.0 too.

**Comment by XiaobingSuper** [comment]
@jay746, thanks, we are tracking it.

**Comment by jay746** [comment]
Thanks @albanD for suggesting a workaround. thanks @XiaobingSuper for tracking an issue.

**Comment by XiaobingSuper** [comment]
This issue will be fixed at next OneDNN release.

**Comment by yanbing-j** [comment]
https://github.com/pytorch/pytorch/pull/63748 has been merged. This issue has been fixed.

**Comment by Jup11ter** [comment]
I have observed a similar crash `RuntimeError: could not create a primitive` on `torch.nn.LazyConv2d` and `torch.nn.Conv1d` in 1.8.0/1.9.0/1.10.0 versions.
It seems like this crash problem happened on many convolution operations.
The good news is that these crashes on these APIs have been fixed in version 1.11.0, and [#63748](https://github.com/pytorch/pytorch/pull/63748) should take effect for these convolution operations.

```python
import torch

in_channels = 105
out_channels = 1
ifm_shape = [1, in_channels, 1, 1]
padding = [8, 5, 9, 4]
groups = 1
kernel_size = [1, 1]
dilation = [1, 1]
stride = [14, 14]

ifm = torch.rand(ifm_shape)
ifm.requires_grad_(True)
padded_ifm = torch.nn.functional.pad(ifm, padding)


op = torch.nn.LazyConv2d(out_channels=out_channels,kernel_size=kernel_size,stride=stride,dilation=dilation,groups=groups)
# op = torch.nn.Conv1d(in_channels=in_channels,out_channels=out_channels,kernel_size=kernel_size,stride=stride,dilation=dilation,groups=groups)


res = op(padded_ifm)

grad_in = torch.rand(res.shape)

res.backward(grad_in)

```
